op-service/httputil/server.go

```diff
@@ -1,30 +1,75 @@
 package httputil
 
 import (
-	"context"
 	"errors"
+	"fmt"
+	"net"
 	"net/http"
+	"sync/atomic"
 )
 
-func ListenAndServeContext(ctx context.Context, server *http.Server) error {
-	errCh := make(chan error, 1)
-	go func() {
-		errCh <- server.ListenAndServe()
-	}()
+// HTTPServer wraps a http.Server, while providing conveniences
+// like exposing the running state and address.
+type HTTPServer struct {
+	listener net.Listener
+	srv      *http.Server
+	closed   atomic.Bool
+}
+
+// HTTPOption applies a change to an HTTP server
+type HTTPOption func(srv *HTTPServer) error
+
+func StartHTTPServer(addr string, handler http.Handler, opts ...HTTPOption) (*HTTPServer, error) {
```

Comment:
Would be good to have some simple unit tests of the `HTTPServer`, e.g. that start it, close it (or shutdown it) and assert that `Closed()` is false before and true after close/shutdown.

Reply:
Added tests for basic/force-shutdown/graceful cases now :+1: